cmd/subctl/reconstruct.go

```diff
@@ -0,0 +1,154 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subctl
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/subctl/internal/cli"
+	"github.com/submariner-io/subctl/internal/constants"
+	"github.com/submariner-io/subctl/internal/exit"
+	"github.com/submariner-io/subctl/internal/restconfig"
+	"github.com/submariner-io/subctl/pkg/broker"
+	"github.com/submariner-io/subctl/pkg/cluster"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	controllerClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var recoverRestConfigProducer = restconfig.NewProducer().WithPrefixedContext("broker").WithDefaultNamespace(constants.OperatorNamespace).
+	WithPrefixedNamespace("broker", constants.DefaultBrokerNamespace)
+
+var (
+	ipsecFile  string
+)
+
+// recoverBrokerInfo represents the reconstruct command.
+var recoverBrokerInfo = &cobra.Command{
+	Use:   "recover-brokerinfo",
+	Short: "Recovers broker-info.subm file from the installed Broker",
+	Run: func(cmd *cobra.Command, args []string) {
+		exit.OnError(recoverBroker(cli.NewReporter()))
+	},
+}
+
+func init() {
+	addReconstructFlags()
+	recoverRestConfigProducer.SetupFlags(recoverBrokerInfo.Flags())
+	rootCmd.AddCommand(recoverBrokerInfo)
+}
+
+func addReconstructFlags() {
+	recoverBrokerInfo.PersistentFlags().StringVar(&ipsecFile, "ipsec-psk-from", "",
+		"import IPsec PSK from existing submariner broker file, like broker-info.subm")
+}
+
+func recoverBroker(status reporter.Interface) error {
+
+	// if --brokerconfig flag provided, get the broker and proceed to get Submariner
+	contextFound, err := recoverRestConfigProducer.RunOnSelectedPrefixedContext("broker",
+		func(brokerCluster *cluster.Info, brokerNamespace string, status reporter.Interface) error {
+			status.Start("Checking if Broker is installed on cluster %s in %s namespace", brokerCluster.Name, brokerNamespace)
+			defer status.End()
+			err := errors.New("error: ")
+			found, err := brokerCluster.HasBroker(brokerNamespace)
+			if err != nil {
+				return err
+			}
+			if found {
+				status.Success("Broker found")
+				status.Start("Checking if Submariner is installed on the same cluster")
+
+				if brokerCluster.Submariner != nil {
+					status.Success("Submariner found")
+					fmt.Printf("getting details")
+					return broker.ReconstructData(brokerCluster, brokerCluster, brokerNamespace, status, ipsecFile)
+				}
+				status.Warning("Submariner is not installed on this cluster. Checking if Submariner is installed on a different cluster")
```

Comment:
This really shouldn't be a warning. Let's start a new status sequence.

```suggestion
				status.Start("Submariner is not installed on this cluster. Checking if Submariner is installed on a different cluster")
```